### PR TITLE
Add Fleet cpu example & add lr scheduler for models

### PR DESCRIPTION
**Pull request by qjing666**
- Add fleet cpu example based on MultiSlotCTR model and support sync/async/geosgd strategies

- Add lr scheduler for users to warm-up or decay learning rate in one code.

### CODE REVIEW

File: python/fleetx/applications/model.py [hunk 1]
```diff
@@ -221,3 +251,31 @@ def load_digital_dataset_from_file(self,
...
+        inputs, loss, startup, main, unique_generator, checkpoints, target = load_program(
+            model_name)
+        self.startup_prog = startup
+        self.main_prog = main
+        self.inputs = inputs
+        self.loss = loss
+        self.checkpoints = checkpoints
+        self.target = target
+
+    def load_multislot_from_file(self,
```
Comment (guru4elephant) [inline python/fleetx/applications/model.py]: load criteo from file

File: python/fleetx/applications/model.py [hunk 2]
```diff
@@ -45,6 +46,35 @@ def startup_program(self):
     def main_program(self):
         return self.main_prog
 
+    def linear_warmup_decay(self, learning_rate, warmup_steps,
```
Comment (guru4elephant) [inline python/fleetx/applications/model.py]: learning rate scheduler is not part of the model.

File: examples/ctr_app.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    fleet.run_server()
+else:
+    fleet.init_worker()
+    exe = fluid.Executor(fluid.CPUPlace())
+    exe.run(fluid.default_startup_program())
+    epochs = 10
+    for epoch in range(epochs):
+        for i, data in enumerate(loader()):
+            print("hi")
+            print(data[0])
```
Comment (guru4elephant) [inline examples/ctr_app.py]: remove this

File: examples/ctr_app.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+configs = X.parse_train_configs()
+role = role_maker.PaddleCloudRoleMaker()
+fleet.init(role)
+model = X.applications.MultiSlotCTR()
+loader = model.load_multislot_from_file('./ctr_data/train_data')
+
+dist_strategy = fleet.DistributedStrategy()
+dist_strategy.mode = 'PS'
+dist_strategy.a_sync = False
+#dist_strategy.a_sync = True
```
Comment (guru4elephant) [inline examples/ctr_app.py]: remove if not used.

File: examples/bert_app.py [hunk 5]
```diff
@@ -40,31 +40,19 @@
 data_loader = model.load_digital_dataset_from_file(
     data_dir='./train_data', vocab_path='./vocab.txt')
 
-place = fluid.CUDAPlace(int(os.environ.get('FLAGS_selected_gpus', 0)))
+learning_rate = X.applications.linear_warmup_decay(configs.lr, 4000, 1000000)
```
Comment (guru4elephant) [inline examples/bert_app.py]: linear_warmup_decay最好放在utils下

File: examples/bert_app.py [hunk 6]
```diff
@@ -34,37 +34,25 @@
 role = role_maker.PaddleCloudRoleMaker(is_collective=True)
 fleet.init(role)
 # load Bert_large / Bert_base model
-#model = X.applications.Bert_large()
-model = X.applications.Bert_base()
+model = X.applications.Bert_large()
+#model = X.applications.Bert_base()
```
Comment (guru4elephant) [inline examples/bert_app.py]: BertLarge()

File: examples/bert_app.py [hunk 7]
```diff
@@ -34,37 +34,25 @@
...
-optimizer = fluid.optimizer.Adam(learning_rate=configs.lr)
+
+optimizer = fluid.optimizer.Adam(learning_rate=learning_rate)
 optimizer = fleet.distributed_optimizer(optimizer, dist_strategy)
 optimizer.minimize(model.loss)
 
-exe = fluid.Executor(place)
-exe.run(fluid.default_startup_program())
+place = fluid.CUDAPlace(int(os.environ.get('FLAGS_selected_gpus', 0)))
+trainer = X.Trainer(place)
```
Comment (guru4elephant) [inline examples/bert_app.py]: How about just MultiGPUTrainer

File: examples/ctr_app.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+optimizer = fleet.distributed_optimizer(optimizer, dist_strategy)
+optimizer.minimize(model.loss)
+
+if fleet.is_server():
+    fleet.init_server()
+    fleet.run_server()
+else:
+    fleet.init_worker()
+    trainer = X.Trainer(fluid.CPUPlace())
```
Comment (guru4elephant) [inline examples/ctr_app.py]: How about just CPUTrainer

File: examples/bert_app.py (hunk 5 above)
Comment (guru4elephant) [reply examples/bert_app.py]: X.utils.linear_warmup_decay(configs.lr, 4000, 1000000)

**Review by guru4elephant — COMMENTED** [review]
please refine API.

File: examples/bert_app.py [hunk 9]
```diff
@@ -33,38 +33,26 @@
...
 dist_strategy.nccl_comm_num = 3
-optimizer = fluid.optimizer.Adam(learning_rate=configs.lr)
+
+optimizer = fluid.optimizer.Adam(learning_rate=learning_rate)
 optimizer = fleet.distributed_optimizer(optimizer, dist_strategy)
 optimizer.minimize(model.loss)
 
-exe = fluid.Executor(place)
-exe.run(fluid.default_startup_program())
+place = fluid.CUDAPlace(int(os.environ.get('FLAGS_selected_gpus', 0)))
```
Comment (guru4elephant) [inline examples/bert_app.py]: 如果是GPUTrainer，是不是在内部声明place即可？

**Review by guru4elephant — APPROVED** [review]
LGTM
